### ISSUE TITLE
Adding more instances to PureLoggerT

### ISSUE THREAD

**Issue by epicallan** [kowainik/co-log#202]
I have a custom monad stack which ends up looking like this when I use it in tests where I want to accumulate the logs and output at the end; this also means `m` in the Monad stack can either be `IO` or `Identity` at various use cases.

```
type TestApp env m a = AppT env (PureLoggerT Message m) a
```
With that background, I found my self needing some more instances over `PureLoggerT` such as MonadIO. I can submit a PR for this if you find it reasonable. 

My PR could effectively be this addition though I also have a few other instances I need but would want to first hear from you on this.

```haskell
instance MonadIO m => MonadIO (PureLoggerT msg m) where
  liftIO = lift . liftIO
```

**Comment by chshersh** [comment]
@epicallan Thanks for suggestion! Adding such `MonadIO` instance sounds like a reasonable thing to do 👍 Feel free to open a PR. Moreover, I believe, it should be even possible to derive this instance instead of writing it manually.

What other instances do you have in mind for `PureLoggerT`?

**Comment by epicallan** [comment]
> it should be even possible to derive this instance instead of writing it manually.

Yes right. I will do that. Thanks

The other instances I have in mind are MonadThrow and MondaFail, which can both be auto derived.

**Comment by chshersh** [comment]
Both `MonadFail` and `MonadThrow` are okay. `MonadThrow` will require to add `exceptions` as a dependency, but as I see, `exceptions` is a boot library since GHC-8.10, so this is fine.

**Comment by chshersh** [comment]
Thanks @epicallan! Resolved by #203

**Comment by epicallan** [comment]
Thanks for the merge and quickly getting on top of this, much appreciated @chshersh
